replication/binlogsyncer.go

```diff
@@ -424,6 +425,15 @@ func (b *BinlogSyncer) GetNextPosition() Position {
 	return b.nextPos
 }
 
+func (b *BinlogSyncer) checkFlavor() {
+	serverVersion := b.c.GetServerVersion()
+	if b.cfg.Flavor != MariaDBFlavor &&
+		strings.Contains(b.c.GetServerVersion(), "MariaDB") {
+		b.cfg.Logger.Errorf("misconfigured flavor (%s) for server %s",
```

Comment:
And I think we can add comment to explain why this is only a log not a failure to caller, like "MariaDB still sends binlog events in a compatible way although some functionalities are lost, so we just log the problem"

Reply:
I've added a code comment. Or did you want to put this in the message instead?